### PR TITLE
feature(CreatePolicy): RHICOMPL-1552 update review page with OSvers

### PR DESCRIPTION
**Pull request by skateman**
Updated the review page in the create policy wizard accordin the 133 mockup. It's displaying the number of systems for each OS minor version separately, sorted in a descending order.

## Secure Coding Practices Checklist GitHub Link
- https://github.com/RedHatInsights/secure-coding-checklist

## Secure Coding Checklist
- [x] Input Validation
- [x] Output Encoding
- [x] Authentication and Password Management
- [x] Session Management
- [x] Access Control
- [x] Cryptographic Practices
- [x] Error Handling and Logging
- [x] Data Protection
- [x] Communication Security
- [x] System Configuration
- [x] Database Security
- [x] File Management
- [x] Memory Management
- [x] General Coding Practices

### CODE REVIEW

File: src/SmartComponents/CreatePolicy/ReviewCreatedPolicy.js [hunk 1]
```diff
@@ -72,11 +60,20 @@ const ReviewCreatedPolicy = ({
...
-                <TextListItem component={TextListItemVariants.dt}>No. of rules</TextListItem>
-                <TextListItem component={TextListItemVariants.dd}>{ rulesCount }</TextListItem>
-                <TextListItem component={TextListItemVariants.dt}>No. of systems</TextListItem>
+                <TextListItem component={TextListItemVariants.dt}>Systems</TextListItem>
                 <TextListItem component={TextListItemVariants.dd}>
-                    { systemsCount > 0 ? systemsCount : <SystemsCountWarning variant="compact" count={ systemsCount } /> }
+                    <TextList component={TextListVariants.dl}>
+                        { Object.keys(osMinorVersions).sort().reverse().map((osMinorVersion) => (
+                            <React.Fragment key={osMinorVersion}>
+                                <TextListItem component={TextListItemVariants.dt}>
```
Comment (skateman) [inline src/SmartComponents/CreatePolicy/ReviewCreatedPolicy.js:68]: This is making the OS version bold which differs from the mockup. If that's not acceptable, I need a little help with what else should I use here :thinking: @bastilian

Comment (bastilian) [reply src/SmartComponents/CreatePolicy/ReviewCreatedPolicy.js:68]: You can add a Patternfly class to the component to set the font weight to normal: https://www.patternfly.org/v4/utilities/text#font-weight That should do the trick.

File: src/SmartComponents/CreatePolicy/ReviewCreatedPolicy.js [hunk 2]
```diff
@@ -72,11 +60,20 @@ const ReviewCreatedPolicy = ({
...
                     </React.Fragment>
                 }
-                <TextListItem component={TextListItemVariants.dt}>No. of rules</TextListItem>
-                <TextListItem component={TextListItemVariants.dd}>{ rulesCount }</TextListItem>
-                <TextListItem component={TextListItemVariants.dt}>No. of systems</TextListItem>
+                <TextListItem component={TextListItemVariants.dt}>Systems</TextListItem>
                 <TextListItem component={TextListItemVariants.dd}>
-                    { systemsCount > 0 ? systemsCount : <SystemsCountWarning variant="compact" count={ systemsCount } /> }
+                    <TextList component={TextListVariants.dl}>
+                        { Object.keys(osMinorVersions).sort().reverse().map((osMinorVersion) => (
```
Comment (bastilian) [inline src/SmartComponents/CreatePolicy/ReviewCreatedPolicy.js:66]: There is a `sortByProp` function that could be used here: https://github.com/RedHatInsights/compliance-frontend/blob/master/src/Utilities/helpers.js#L3

Comment (skateman) [reply src/SmartComponents/CreatePolicy/ReviewCreatedPolicy.js:66]: But I'm not sorting by prop, I'm just sorting an array :thinking:

File: src/SmartComponents/CreatePolicy/ReviewCreatedPolicy.js (hunk 1 above)
Comment (skateman) [reply src/SmartComponents/CreatePolicy/ReviewCreatedPolicy.js:68]: Nope, `dt` ignores the `pf-u-font-weight-normal` class :disappointed:

File: src/SmartComponents/CreatePolicy/ReviewCreatedPolicy.js (hunk 2 above)
Comment (bastilian) [reply src/SmartComponents/CreatePolicy/ReviewCreatedPolicy.js:66]: @skateman ... it should probably be named `sortArrayByItemProp` or something. It takes a property name of an array item as an argument, and returns a sorting function that can be passed to the `sort` function.

Comment (skateman) [reply src/SmartComponents/CreatePolicy/ReviewCreatedPolicy.js:66]: Yeah, I get that, but I'm not sorting an array of objects, I'm sorting an array of strings :thinking:

File: src/SmartComponents/CreatePolicy/ReviewCreatedPolicy.js [hunk 3]
```diff
@@ -14,56 +14,44 @@ import gql from 'graphql-tag';
...
     const { data, error, loading } = useQuery(REVIEW, { variables: { benchmarkId } });
 
     if (error) { return error; }
 
     if (loading) { return <Spinner/>; }
 
-    const benchmark = data.benchmark;
+    const { benchmark: { osMajorVersion } } = data;
+
+    const osMinorVersions = systems.reduce((acc, { node: { osMinorVersion } }) => ({
```
Comment (bastilian) [inline src/SmartComponents/CreatePolicy/ReviewCreatedPolicy.js:38]: Why not use the `systemsCount` that has the counts already accumulated?

Comment (skateman) [reply src/SmartComponents/CreatePolicy/ReviewCreatedPolicy.js:38]: Yes, but not grouped by OS version, which I need.

Comment (bastilian) [reply src/SmartComponents/CreatePolicy/ReviewCreatedPolicy.js:38]: That it is exactly what it is. it returns items with the osMinorVersion and the count of systems selected for that.
There is also a form state for the the profile to get the selected major version, that all that should be needed in order to construct the list of systems, or?

Comment (skateman) [reply src/SmartComponents/CreatePolicy/ReviewCreatedPolicy.js:38]: The `systemsCount` is a number, while I need a number of systems for each minor version.

**Comment by vkrizan** [comment]
@skateman can you please also drop the "Tailoring for" within the Rules step, unless that creates too much of a conflict? Thank you.

**Comment by skateman** [comment]
> @skateman can you please also drop the "Tailoring for" within the Rules step, unless that creates too much of a conflict? Thank you.

Shouldn't that be rather done with the tabs coming?

File: src/SmartComponents/CreatePolicy/ReviewCreatedPolicy.js (hunk 3 above)
Comment (vkrizan) [reply src/SmartComponents/CreatePolicy/ReviewCreatedPolicy.js:38]: @skateman Have you checked `osMinorVersionCounts` state variable? It should have everything you need.

Here is the type signature:
```
    osMinorVersionCounts: propTypes.arrayOf(propTypes.shape({
        osMinorVersion: propTypes.number,
        count: propTypes.number
    }))
```

The array is already sorted in ascending order. You can call `reverse()` to render the final review list.

**Review by vkrizan — COMMENTED** [review]
Works as expected. Would you mind refactoring it a bit? Thank you!

**Comment by vkrizan** [comment]
> Shouldn't that be rather done with the tabs coming?

Of course.

**Review by vkrizan — APPROVED** [review]
Thank you!

**Pull request merged**